### PR TITLE
Add SWIG and Travis CI

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This PR adds SWIG required for building tesseract_python. Also it fixes some of the packages depends so things build in the correct order.

Update: Also it adds Travis CI for linux and Windows. I excluded kinetic build because there is a bug in catkin that requires at least one catkin package. It will not build a workspace of all pure cmake packages even if they have package.xml files. This has been fixed in melodic.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  global:
+    - CCACHE_DIR=$HOME/.ccache
+    - ROS_REPO=ros
+    - NOT_TEST_INSTALL=true
+    - ADDITIONAL_DEBS="autotools-dev automake libpcre3-dev bison"
+
+jobs:
+  include:
+    - os: linux
+      dist: xenial
```
Comment (mpowelson) [inline .travis.yml:26]: Should this be bionic?

File: swig/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.5.0)
+
+include(ExternalProject)
+
+find_package(SWIG QUIET)
```
Comment (mpowelson) [inline swig/CMakeLists.txt:5]: You should add SWIG to the readme under required. Also it looks like I missed libccd

Comment (Levi-Armstrong) [reply swig/CMakeLists.txt:5]: updated

File: .travis.yml (hunk 1 above)
Comment (Levi-Armstrong) [reply .travis.yml:26]: yep

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - ADDITIONAL_DEBS="autotools-dev automake libpcre3-dev bison"
+
+jobs:
+  include:
+    - os: linux
+      dist: bionic
+      language: cpp
+      env:
+      - ROS_DISTRO="melodic"
+      - BADGE=xenial
```
Comment (mpowelson) [inline .travis.yml]: Does this matter?

Comment (Levi-Armstrong) [reply .travis.yml]: The BADGE tag will be used for creating badges for individual CI builds.

**Comment by Levi-Armstrong** [comment]
@johnwason I am trying to get swig to build on windows but it uses automake and autotools and I am not sure how to install these tools on windows. Have install swig on windows?

**Comment by johnwason** [comment]
I have never been able to get SWIG to build on windows. I think it needs to be built with MinGW. My suggestion is to use the binaries available on the SWIG website.

http://www.swig.org/download.html

http://prdownloads.sourceforge.net/swig/swigwin-4.0.1.zip

See the Robot Raconteur travis.yml for an example:

https://github.com/robotraconteur/robotraconteur/blob/master/.travis.yml

**Comment by Levi-Armstrong** [comment]
@johnwason I almost have swig working in the windows ci. The only think I cannot figure out is how to move the swig.exe to the install bin director and rename it. Do you have any thoughts?

**Comment by johnwason** [comment]
I think you will have to put custom `install` directives in the CMakeLists.txt file. As long as the relative directory structure is unchanged it should be fine.

**Comment by Levi-Armstrong** [comment]
This is what I have so far but the it does not like my install command. I tested it on a windows machine in the command prompt which worked but not sure why it is not working here.

```
ExternalProject_Add(swig
      URL               https://sourceforge.net/projects/swig/files/swigwin/swigwin-4.0.1/swigwin-4.0.1.zip/download
      CONFIGURE_COMMAND ""
      BUILD_COMMAND     ""
      INSTALL_COMMAND   if not exist ${CMAKE_INSTALL_PREFIX}/bin mkdir ${CMAKE_INSTALL_PREFIX}/bin && mv ${CMAKE_BINARY_DIR}/swig-prefix/src/swig/swig.exe ${CMAKE_INSTALL_PREFIX}/bin/swig4.0.exe
    )
    )
```

**Comment by johnwason** [comment]
Oh, I forgot that I used chocolatey to install on Windows with the more recent versions. Add:

`choco install swig`

to the setup commands in travis.yml. I am not sure if rosdep has this configured for automatic install. 

Disable the swig package in tesseract_ext for Windows and use choco instead.

**Comment by Levi-Armstrong** [comment]
Ah, Thank you.

**Comment by Levi-Armstrong** [comment]
@mpowelson Travis CI is now passing.

**Pull request merged**